### PR TITLE
MGMT-3062 Allow cluster install progress updates for cluster installing and finalizing state

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
/hold
testing

File: internal/cluster/statemachine.go [hunk 1]
```diff
@@ -280,6 +271,17 @@ func NewClusterStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostRefreshCluster(statusInfoError),
 	})
 
+	for _, state := range []stateswitch.State{
+		stateswitch.State(models.ClusterStatusInstalling),
+		stateswitch.State(models.ClusterStatusFinalizing)} {
+		sm.AddTransition(stateswitch.TransitionRule{
+			TransitionType:   TransitionTypeUpdateInstallationProgress,
+			SourceStates:     []stateswitch.State{state},
```
Comment (filanov) [inline internal/cluster/statemachine.go:279]: you don't need a for, you the parameter is already a list so just assign it directly.

Comment (nmagnezi) [reply internal/cluster/statemachine.go:279]: I did this to match the `SourceState` with `DestinationState`

**Comment by nmagnezi** [comment]
/unhold
Captured update from assisted-service pod log [here](https://gist.github.com/nmagnezi/dd3089fc6f96c9563aadc6e09f817b60)

File: internal/cluster/statemachine.go (hunk 1 above)
Comment (filanov) [reply internal/cluster/statemachine.go:279]: oh right

**Comment by filanov** [comment]
Can you please add an example of the output here?

**Comment by nmagnezi** [comment]
> Can you please add an example of the output here?

@filanov, sure
```golang
time="2020-12-20T08:57:12Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T08:57:12.981Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=5fa0aa3d-5c77-4366-b4cc-3a536010c8aa
time="2020-12-20T08:57:18Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: downloading update progress_progress_updated_at 2020-12-20T08:57:18.656Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=b24f1d58-7fee-41e5-8ff0-db169ebfbc81
time="2020-12-20T08:57:40Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: downloading update progress_progress_updated_at 2020-12-20T08:57:40.979Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=72df886b-2271-45b8-9e33-065b5a261960
time="2020-12-20T08:58:10Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T08:58:10.976Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=82a5d0a4-c621-498f-910f-faa5bf894565
time="2020-12-20T08:58:40Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T08:58:40.980Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=bd2f38d7-54e6-42f4-b7c7-f5b5deb0e368
time="2020-12-20T08:59:10Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T08:59:10.976Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=1d4e92cf-abda-43d2-8607-beea4e7564ef
time="2020-12-20T08:59:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T08:59:40.977Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=5d377668-80f0-4bf9-8650-f7a11e7a9c4c
time="2020-12-20T09:00:10Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T09:00:10.976Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=8bda9647-b4bb-47df-91fa-9f7dd42a368f
time="2020-12-20T09:00:43Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: downloading update progress_progress_updated_at 2020-12-20T09:00:43.103Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=d24a5930-c818-4f02-b972-65deaa0fd432
time="2020-12-20T09:01:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 74% complete progress_progress_updated_at 2020-12-20T09:01:11.658Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=4ce221ec-2bb4-4703-b8b0-21816d02d5d7
time="2020-12-20T09:01:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:01:41.068Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=d77dfb12-e8ff-4495-afc3-ffe08e45a7ab
time="2020-12-20T09:02:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:02:11.451Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=e42c1a8f-2237-4a7e-aea3-3af91a9cf98f
time="2020-12-20T09:02:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:02:41.023Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=f66f7eba-d5d2-498a-9020-4e0fd9099d3c
time="2020-12-20T09:03:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:03:11.091Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=cc8e9f6f-e603-42fc-b67b-e586094675b6
time="2020-12-20T09:03:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:03:40.997Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=0c1f9a9f-8bd8-442f-92e2-efcddacf0dee
time="2020-12-20T09:04:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:04:10.982Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=3e156f37-6eea-4a9d-9e42-bc078320df03
time="2020-12-20T09:04:40Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:04:40.982Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=a13b5d8b-b31f-4395-ba8d-b99252308818
time="2020-12-20T09:05:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:05:10.983Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=22b3780c-08c2-4b4c-b4a1-b7a316f32204
time="2020-12-20T09:05:40Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T09:05:40.980Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=d99d73e0-b5cc-4281-bb06-4b0cdfa74913
time="2020-12-20T09:06:10Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message: progress_progress_updated_at 2020-12-20T09:06:10.980Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=cfccb977-f55e-4562-956c-9bbde0d9ef43
time="2020-12-20T09:06:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 97% complete progress_progress_updated_at 2020-12-20T09:06:41.035Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=505bd535-324e-4800-a90c-0976af6c9178
time="2020-12-20T09:07:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:07:10.981Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=8ab09744-4a04-4679-861e-0782e5cc691e
time="2020-12-20T09:07:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:07:40.987Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=a1838ffd-446e-408e-9bb3-ec74d183952a
time="2020-12-20T09:08:22Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:08:22.191Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=26858 pkg=cluster-state request_id=9eb8fbdd-b87b-46ed-9ca5-86e011ac5d13
time="2020-12-20T09:08:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:08:40.982Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=854bf7d0-ae87-492f-aed3-5c82314a8d02
time="2020-12-20T09:09:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:09:10.983Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=a40172a7-2662-4a49-9f04-32ac3bb3faf7
time="2020-12-20T09:09:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:09:40.982Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=96ddea2d-03fd-4085-9dd2-f5ac6a94acac
time="2020-12-20T09:10:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:10:10.981Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=1d40249d-9a28-4bbd-bd16-7626007a2992
time="2020-12-20T09:10:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:10:41.001Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=b84ee1c2-5e4a-45a7-90d5-74c223c87d1b
time="2020-12-20T09:11:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:11:10.990Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=bf3d1ddf-7aa6-4ed2-be9a-65b65a5cc43b
time="2020-12-20T09:11:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:11:40.984Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=3e08cb4b-444e-4816-a74e-559253dd8e18
time="2020-12-20T09:12:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:12:10.988Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=ee56a5da-1244-405d-ac9b-99c9b0d1157b
time="2020-12-20T09:12:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:12:40.983Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=5b6e0bad-d5cc-410f-874d-83484f2a420f
time="2020-12-20T09:13:19Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:13:19.337Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=f48331cb-a091-49e2-a931-7b79b4295b83
time="2020-12-20T09:13:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: an unknown error has occurred: MultipleErrors progress_progress_updated_at 2020-12-20T09:13:40.986Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=b4f07421-ce60-4b04-a82d-deb7a1fedd80
time="2020-12-20T09:14:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 98% complete progress_progress_updated_at 2020-12-20T09:14:10.985Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=8421ba9e-ae9f-4c3d-9ab1-b942255cb00f
time="2020-12-20T09:14:40Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Working towards 4.6.4: 100% complete progress_progress_updated_at 2020-12-20T09:14:40.981Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=0c513b2e-1b26-44bb-9173-106dd51a9608
time="2020-12-20T09:15:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:15:10.986Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=c8c2564d-7768-4697-b61f-7bfa108f30fd
time="2020-12-20T09:15:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:15:41.023Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=bf8e7403-3107-4ced-9696-ac3b625e0220
time="2020-12-20T09:16:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:16:10.990Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=64cfddb0-17b7-421b-a98c-a7f8795c2d61
time="2020-12-20T09:16:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:16:41.008Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=94b3a938-1af2-4b03-8130-789c4905b480
time="2020-12-20T09:17:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:17:11.090Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=212afdaf-5514-486e-b20f-a824392142d2
time="2020-12-20T09:17:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:17:40.993Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=2e623fda-d970-41a3-918d-895b3b6e16c6
time="2020-12-20T09:18:11Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:false , message:Unable to apply 4.6.4: the cluster operator authentication is degraded progress_progress_updated_at 2020-12-20T09:18:10.979Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=12c176e3-1f0f-4d58-a93c-673810aa3cc3
time="2020-12-20T09:18:41Z" level=info msg="cluster 976d44b8-0ca7-4a44-b569-b873a4af4e9c has been updated with the following updates [progress_progress_info Cluster version is available:true , message:Done applying 4.6.4 progress_progress_updated_at 2020-12-20T09:18:40.984Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=976d44b8-0ca7-4a44-b569-b873a4af4e9c go-id=20033 pkg=cluster-state request_id=ebf4ef06-6cba-4e0d-8afe-4c43eef5f7cf
```

**Review by filanov — APPROVED** [review]
We can start like this and in worst case we can find a way to reduce the logs.

**Comment by filanov** [comment]
@nmagnezi just make sure that the internal is not every second.

**Pull request merged**